src/crimson/os/seastore/transaction_manager.h

```diff
@@ -191,10 +223,16 @@ class TransactionManager : public ExtentCallbackInterface {
   }
 
   template <typename T>
-  base_iertr::future<TCachedExtentRef<T>> read_pin(
+  base_iertr::future<maybe_indirect_extent_t<T>> read_pin(
     Transaction &t,
     LBAMappingRef pin)
   {
+    bool is_clone = pin->is_clone();
+    std::optional<indirect_info_t> maybe_indirect_info;
+    if (pin->is_indirect()) {
```

Comment:
If the extent was cloned but the snapshot was deleted later on, I think for the remaining head, the pin keeps indirect but it might not be a clone (refcnt == 1), is it correct?

Reply:
Ah, yes, that's right.